arcgis-ios-sdk-samples/Edit data/Edit with branch versioning/EditWithBranchVersioningViewController.swift

```diff
@@ -0,0 +1,542 @@
+                    guard let self = self else { return }
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
+                        // Reload feature table with the new version.
+                        self.featureLayer.featureTable?.load()
+                        self.currentVersionName = branchVersionName
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Apply local edits to the geodatabase.
+    ///
+    /// - Parameters:
+    ///   - geodatabase: The geodatabase to apply edits.
+    ///   - completion: An optional closure to execute after edits are applied.
+    func applyLocalEdits(geodatabase: AGSServiceGeodatabase, completion: (() -> Void)? = nil) {
+        if geodatabase.hasLocalEdits() {
+            SVProgressHUD.show(withStatus: "Applying local edits…")
+            geodatabase.applyEdits { _, _ in
+                SVProgressHUD.dismiss()
+                completion?()
+            }
+        } else {
+            completion?()
+        }
+    }
+    
+    /// Undo local edits on the geodatabase.
+    ///
+    /// - Parameters:
+    ///   - geodatabase: The geodatabase to discard edits.
+    ///   - completion: An optional closure to execute after edits are undone.
+    func undoLocalEdits(geodatabase: AGSServiceGeodatabase, completion: (() -> Void)? = nil) {
+        if geodatabase.hasLocalEdits() {
+            SVProgressHUD.show(withStatus: "Discarding local edits…")
+            geodatabase.undoLocalEdits { _ in
+                SVProgressHUD.dismiss()
+                completion?()
+            }
+        } else {
+            completion?()
+        }
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func createBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        // Clear selection before creating a new branch version.
+        clearSelection()
+        mapView.callout.dismiss()
+        chooseVersionAccessPermission(sender) { permission in
+            self.createBranchAlert(permission: permission) { [weak self] parameters in
+                guard let self = self else { return }
+                self.createVersion(geodatabase: self.serviceGeodatabase, with: parameters) { [weak self] result in
+                    guard let self = self else { return }
+                    switch result {
+                    case .success(let versionName):
+                        self.existingVersionNames.append(versionName)
+                        // Switch to the new version after it is created.
+                        self.switchVersion(geodatabase: self.serviceGeodatabase, to: versionName)
+                    case .failure(let error as NSError):
+                        // Provide additional error reason to users if there is any.
+                        let errorMessage = error.localizedDescription + (error.localizedFailureReason ?? "")
```

Comment:
For example, when I create another version branch that has the same name as an existing branch, the service gives me an error "Error during createVersion operation", while its `localizedFailureReason` explains "The version already exists".

Should we incorporate this more detailed failure reason? I feel it might be too generalized to only have the first part (the `localizedDescription`, which is also the message of a `presentAlert`), as there are several types of error for `createVersion`. But I didn't find a proper way to do so. 🤔

Reply:
Since `localizedFailureReason` is optional, you could use `error.localizedFailureReason ?? error.localizedDescription ` as the message.

Reply:
Hmm I decided to stick with the initial change you suggested. Another error message for invalid branch name look like this

![image](https://user-images.githubusercontent.com/9660181/92043712-6d3b3a80-ed31-11ea-9d73-692b1ecf6733.png)

where the additional message does not provide anything meaningful besides an error code. The error messages given by the core is not consistent, thus I'll stick with the `NSLocalizedDescription`/`localizedDescription`.